Export Caffe2 FC and BatchMatMul to C10 for Bert ONNX model

xiaomengy opened the pull request:
Summary: Export Caffe2 FC and BatchMatMul to C10 for Bert ONNX model

Differential Revision: D15057161

houseroad reviewed (approved):
LGTM.

Nit: add pytorch==>onnx==>caffe2 end to end tests like the following case: https://github.com/pytorch/pytorch/blob/master/test/onnx/test_pytorch_onnx_caffe2.py#L1374

dzhulgakov commented on caffe2/operators/fully_connected_op.cc:
```diff
@@ -237,3 +243,23 @@ REGISTER_GRADIENT(FCTransposed, GetFCGradient);
+C10_REGISTER_CAFFE2_OPERATOR_CPU(
+    FC,
+    "_caffe2::FC("
+    "    Tensor[] inputs_list,"
```
why do you bind this op and other ops with tensor list as input? why not create individual inputs?

BIT-silence replied:
Fixed

xiaomengy committed:
Export Caffe2 FC and BatchMatMul to C10 for Bert ONNX model (#19644)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/19644

Export Caffe2 FC and BatchMatMul to C10 for Bert ONNX model

Reviewed By: houseroad

Differential Revision: D15057161

fbshipit-source-id: ce05caa34ed0274c8c6f25bfba3dc1be15bc9f72
